### PR TITLE
Infer .swiftdoc and .swiftsourceinfo paths from .swiftmodule path in a given outputFileMap

### PR DESCRIPTION
**Pull request by nkcsgexi**
This allows the -merge-module action to find and import those partial entities.

rdar://70494813

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please test

File: Sources/SwiftDriver/Driver/OutputFileMap.swift [hunk 1]
```diff
@@ -42,7 +42,17 @@ public struct OutputFileMap: Hashable, Codable {
   }
 
   public func existingOutput(inputFile: VirtualPath, outputType: FileType) -> VirtualPath? {
-    entries[inputFile]?[outputType]
+    switch outputType {
+    case .swiftDocumentation, .swiftSourceInfoFile:
+      // If the output map has specific entry for these entities, use that.
+      if let path = entries[inputFile]?[outputType] {
+        return path
+      }
```
Comment (DougGregor) [inline Sources/SwiftDriver/Driver/OutputFileMap.swift:50]: How about making this the first check in the function, and then the `switch` only contains the fallback options or returns `nil`?

**Comment by nkcsgexi** [comment]
@swift-ci please test

**Review by DougGregor — APPROVED** [review]

File: Sources/SwiftDriver/Driver/OutputFileMap.swift (hunk 1 above)
Comment (nkcsgexi) [reply Sources/SwiftDriver/Driver/OutputFileMap.swift]: Sounds good, Fixed.

**Comment by nkcsgexi** [comment]
@swift-ci please test

**Pull request merged**